### PR TITLE
Add script to reload events if pending.data changed

### PR DESCRIPTION
**Pull request by jspricke**
This needs the Python watchdog library

### CODE REVIEW

File: scripts/vit-watchdog [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self):
+        super().__init__(patterns=["*/pending.data"], ignore_directories=True)
+
+    def on_modified(self, event):
+        if p:
+            p.send_signal(signal.SIGUSR1)
+
+
+observer = Observer()
+observer.schedule(VitEventHandler(), expanduser("~/.local/share/task"))
```
Comment (thehunmonkgroup) [inline scripts/vit-watchdog]: Would be better if this was not hard-coded, not all people store their task data here.

File: scripts/vit-watchdog [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+observer.start()
+
+p = Popen("/usr/bin/vit")
```
Comment (thehunmonkgroup) [inline scripts/vit-watchdog]: Same with this, not everyone has their vit executable here.

File: scripts/vit-watchdog [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
+from os.path import expanduser
+from subprocess import Popen, signal
+
+from watchdog.events import PatternMatchingEventHandler
```
Comment (thehunmonkgroup) [inline scripts/vit-watchdog]: Perhaps if watchdog is required, and is not a standard module available in a vit environment, that there should be some graceful failure here, to inform the user to install the package?

**Review by thehunmonkgroup — CHANGES_REQUESTED** [review]

File: scripts/vit-watchdog (hunk 1 above)
Comment (jspricke) [reply scripts/vit-watchdog]: done

File: scripts/vit-watchdog (hunk 3 above)
Comment (jspricke) [reply scripts/vit-watchdog]: done

File: scripts/vit-watchdog (hunk 2 above)
Comment (jspricke) [reply scripts/vit-watchdog]: done

**Comment by thehunmonkgroup** [comment]
Changes look good. I think it would be helpful to add some documentation, what are your thoughts about the best location? Could be added as a comment at the top of the script, or in some of the other existing markdown doc in the repo?

**Comment by scottkosty** [comment]
Note that Taskwarrior 3.0 (released in March) does not support pending.data. It now uses a single database (taskchampion.sqlite3) rather than separate databases for pending, completed, etc.

**Comment by thehunmonkgroup** [comment]
> Note that Taskwarrior 3.0 (released in March) does not support pending.data

Yep, I saw this. My understanding of the upgrade is that VIT should work seamlessly with 3.x, as the CLI interface has not changed, only the storage mechanism.

I have not personally upgraded to 3.x, and will not do so until it's matured a bit more. I suspect I am not alone in this decision.

That being said, if SQLite is being used as the local storage mechanism, I would think this kind of file listener approach could still work?

**Comment by jspricke** [comment]
> Could be added as a comment at the top of the script, or in some of the other existing markdown doc in the repo?

Added a comment.

> Yep, I saw this. My understanding of the upgrade is that VIT should work seamlessly with 3.x, as the CLI interface has not changed, only the storage mechanism.

Seems to work fine.

> I have not personally upgraded to 3.x, and will not do so until it's matured a bit more. I suspect I am not alone in this decision.

Same here. We also have to think about what to do with the Debian package as it will be a bit of work to get all that Rust stuff in.

> That being said, if SQLite is being used as the local storage mechanism, I would think this kind of file listener approach could still work?

Sadly it does not work as running `task` creates a write event to the database. @scottkosty do you know if that write event can be dropped?

File: scripts/vit-watchdog [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+#
+# Simple wrapper to reload vit upon task changes
```
Comment (thehunmonkgroup) [inline scripts/vit-watchdog]: The description feels a little vague.

Perhaps something more like this?

```python
# Watchdog script for vit -- automatically reloads vit when task data changes.
# Monitors the pending.data file and sends a SIGUSR1 signal to vit to trigger a refresh.
```

**Comment by thehunmonkgroup** [comment]
> Sadly it does not work as running `task` creates a write event to the database.

I understand the purpose of this PR, as a simple solution to handle automatic updates, which is why I'll commit it when it's ready. However, there's already a documented method for doing auto-refresh: https://github.com/vit-project/vit/blob/2.x/CUSTOMIZE.md#auto-refreshing-vits-interface -- this method will continue to work in Taskwarrior 3.x

File: scripts/vit-watchdog (hunk 4 above)
Comment (thehunmonkgroup) [reply scripts/vit-watchdog]: Also, it just occurred to me that the comment should mention that this will ONLY work with Taskwarrior < 3.0

**Comment by scottkosty** [comment]
> > That being said, if SQLite is being used as the local storage mechanism, I would think this kind of file listener approach could still work?
> 
> Sadly it does not work as running `task` creates a write event to the database. @scottkosty do you know if that write event can be dropped?

Unfortunately I don't know anything about 3.x under the hood.

File: scripts/vit-watchdog (hunk 4 above)
Comment (jspricke) [reply scripts/vit-watchdog]: done.

File: scripts/vit-watchdog [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+#
+# Watchdog script for vit -- automatically reloads vit when task data changes.
+# Monitors the pending.data file and sends a SIGUSR1 signal to vit to trigger a refresh.
```
Comment (thehunmonkgroup) [inline scripts/vit-watchdog:4]: I mentioned this before, we definitely want to add a note making it clear this script ONLY works with Taskwarrior < 3.0.

**Review by thehunmonkgroup — CHANGES_REQUESTED** [review]

File: scripts/vit-watchdog (hunk 5 above)
Comment (jspricke) [reply scripts/vit-watchdog:4]: done.

**Pull request merged**